dev/io.openliberty.microprofile.reactive.streams.operators30.internal/test/src/io/openliberty/microprofile/reactive/streams/test/SimplePubSubTest.java

```diff
@@ -0,0 +1,229 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.reactive.streams.test;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.eclipse.microprofile.reactive.streams.operators.CompletionRunner;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+import org.junit.Test;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ *
+ */
+public class SimplePubSubTest extends AbstractReactiveUnitTest {
+
+    @Test
+    public void testPubSub() {
+        MyPublisher publisher = new MyPublisher();
+        MySubscriber<String> subscriber = new MySubscriber<String>();
+
+        PublisherBuilder<String> pBuilder = ReactiveStreams.fromPublisher(publisher);
+        SubscriberBuilder<String, Void> sBuilder = ReactiveStreams.fromSubscriber(subscriber);
+
+        CompletionRunner<Void> runner = pBuilder.to(sBuilder);
+        CompletionStage<Void> result = runner.run();
+
+        int loops = 0;
+
+        while (!subscriber.isComplete() && loops++ < 10 * 60 * 5) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+
+        List<String> messages = subscriber.getMessages();
+
+        assertTrue(messages.contains("one"));
+        assertTrue(messages.contains("two"));
+        assertTrue(messages.contains("three"));
+    }
+
+    private static class MyPublisher implements Publisher<String> {
```

Comment:
We discussed before that these custom publisher and subscriber classes are a bit hokey, they could do with some cleanup to make them simpler and some javadoc to explain what they're trying to do.